modules/core/nuxeo-core-bulk/src/main/java/org/nuxeo/ecm/core/bulk/computation/BulkScrollerComputation.java

```diff
@@ -284,20 +286,21 @@ protected void updateStatusAsScrolling(ComputationContext context, String comman
                 BulkCodecs.getStatusCodec().encode(delta));
     }
 
-    protected void updateStatusAfterScroll(ComputationContext context, String commandId, String errorMessage) {
-        updateStatusAfterScroll(context, commandId, 0, errorMessage, false);
+    protected void updateStatusAfterScroll(ComputationContext context, String commandId, String errorMessage,
+            int errorCode) {
+        updateStatusAfterScroll(context, commandId, 0, errorMessage, errorCode, false);
     }
 
     protected void updateStatusAfterScroll(ComputationContext context, String commandId, long documentCount,
             boolean limited) {
-        updateStatusAfterScroll(context, commandId, documentCount, null, limited);
+        updateStatusAfterScroll(context, commandId, documentCount, null, 0, limited);
     }
 
     protected void updateStatusAfterScroll(ComputationContext context, String commandId, long documentCount,
-            String errorMessage, boolean limited) {
+            String errorMessage, int errorCode, boolean limited) {
         BulkStatus delta = BulkStatus.deltaOf(commandId);
         if (errorMessage != null) {
-            delta.inError(errorMessage);
+            delta.inError(errorMessage, errorCode);
```

Comment:
I would say that errorCode could be an `Integer`, and call the inError method as below:
```java
delta.inError(errorMessage, requireNonNullElse(errorCode, SC_INTERNAL_SERVER_ERROR));
```
What do you think?